streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -621,7 +628,7 @@ void suspend(final boolean clean,
             try {
                 commit(false, partitionTimes);
             } finally {
-                if (eosEnabled) {
+                if (eosEnabled && !eosBetaEnabled) {
```

Comment:
For line 635 below: again this is not for this PR, but do we really need to close task-producer upon task.suspend? or could we just close it in task.close.

Reply:
Agreed. Not sure why we decided to close the task-producer. But it seems sufficient to only commit a transaction and start a new transaction in `suspend()` / `resume()`